File: packages/emitter/testcases/types/tuples-arity/TuplesArity.ts
// Test tuple types with >7 elements (requires .NET ValueTuple nesting)

// 8-element tuple: ValueTuple<T1..T7, ValueTuple<T8>>
export type T8 = [
  number,
  number,
  number,
  number,
  number,
  number,
  number,
  number,
];

// 9-element tuple: ValueTuple<T1..T7, ValueTuple<T8, T9>>
export type T9 = [
  number,
  number,
  number,
  number,
  number,
  number,
  number,
  number,
  number,
];

// Tuple literals with 8+ elements
export const t8: T8 = [1, 2, 3, 4, 5, 6, 7, 8];
export const t9: T9 = [1, 2, 3, 4, 5, 6, 7, 8, 9];

// Function returning 8-element tuple
export function makeT8(): T8 {
  return [10, 20, 30, 40, 50, 60, 70, 80];
}

// Function with 8-element tuple parameter
export function sumT8(t: T8): number {
  return t[0] + t[1] + t[2] + t[3] + t[4] + t[5] + t[6] + t[7];
}

File: packages/emitter/testcases/types/tuples-arity/TuplesArity.cs
namespace TestCases.types.tuplesarity
{
        public static class TuplesArity
        {
            // type T8 = global::System.ValueTuple<double, double, double, double, double, double, double, global::System.ValueTuple<double>>

            // type T9 = global::System.ValueTuple<double, double, double, double, double, double, double, global::System.ValueTuple<double, double>>

            public static readonly global::System.ValueTuple<double, double, double, double, double, double, double, global::System.ValueTuple<double>> t8 = (1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0);

            public static readonly global::System.ValueTuple<double, double, double, double, double, double, double, global::System.ValueTuple<double, double>> t9 = (1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0);

            public static global::System.ValueTuple<double, double, double, double, double, double, double, global::System.ValueTuple<double>> makeT8()
                {
                return (10.0, 20.0, 30.0, 40.0, 50.0, 60.0, 70.0, 80.0);
                }

            public static double sumT8(global::System.ValueTuple<double, double, double, double, double, double, double, global::System.ValueTuple<double>> t)
                {
                return t[0] + t[1] + t[2] + t[3] + t[4] + t[5] + t[6] + t[7];
                }
        }
}